src/Morphir/TestCoverage/Backend.elm

```diff
@@ -0,0 +1,98 @@
+import Morphir.IR.Name exposing (Name)
+import Morphir.IR.NodeId as NodeId exposing (..)
+import Morphir.IR.Package exposing (PackageName)
+import Morphir.IR.Value as Value
+
+
+type alias Coverage =
+    { numberOfBranches : Int
+    , numberOfCoveredBranches : Int
+    }
+
+
+type alias TestCoverageResult =
+    AssocDict.Dict NodeID Coverage
+
+
+createNodeId : FQName -> NodeID
+createNodeId fqn =
+    fqn |> ValueID
+
+
+calculateNumberOfCoveredBranches : List ( Branch ta va, List TestCase ) -> Int
+calculateNumberOfCoveredBranches branchCoverageResult =
+    branchCoverageResult
+        |> List.filter
+            (\item ->
+                not (item |> Tuple.second |> List.isEmpty)
+            )
+        |> List.length
+
+
+getBranchCoverage : ( PackageName, ModuleName ) -> IR -> TestSuite -> Module.Definition ta va -> TestCoverageResult
+getBranchCoverage ( packageName, moduleName ) ir testSuite moduleDef =
+    moduleDef.values
+        |> Dict.toList
+        |> List.map
+            (\( valueName, accesscontrolledValueDef ) ->
+                let
+                    currentFQN =
+                        FQName.fQName packageName moduleName valueName
+
+                    valueTestCases =
+                        testSuite
+                            |> Dict.get currentFQN
+                            |> Maybe.withDefault []
+
+                    valueDef =
+                        accesscontrolledValueDef.value.value
+                in
+                ( createNodeId currentFQN
+                , valueTestCases
+                    |> BranchCoverage.assignTestCasesToBranches ir valueDef
+                    |> (\lstOfBranchAndCoveredTestCases ->
+                            { numberOfBranches = lstOfBranchAndCoveredTestCases |> List.length
+                            , numberOfCoveredBranches = calculateNumberOfCoveredBranches lstOfBranchAndCoveredTestCases
+                            }
+                       )
+                )
+            )
+        |> AssocDict.fromList
+
+
+
+--|> BranchCoverage.valueBranches True
+--|> (\lstOfBranches ->
+--        ( createNodeId pckgName modName valName
+--        , { numberOfBranches = lstOfBranches |> List.length
+--          , numberOfCoveredBranches = getAssignedTestCaseToBranch () accesscontrolledValueDef.value.value testSuite
+--          }
+--        )
+--   )
+
+
+encodeTestCoverageResult : TestCoverageResult -> Encode.Value
+encodeTestCoverageResult testCoverageResult =
+    testCoverageResult
+        |> AssocDict.toList
+        |> List.map
+            (\( nodeId, coverage ) ->
+                ( NodeId.nodeIdToString nodeId
+                , Encode.object
+                    [ ( "numberOfBranches", Encode.int coverage.numberOfBranches )
+                    , ( "numberOfCoveredBranches", Encode.int coverage.numberOfCoveredBranches )
+                    ]
+                )
+            )
+        |> Encode.object
```

Comment:
While this might seem overkill for a single function but the standard location for this would be in a separate module `Morphir.TestCoverage.Backend.Codec`.

Reply:
separated